### DISCUSSION TITLE
select count with raw query gives n as suffix

### DISCUSSION THREAD

**Discussion by mukul-123** [prisma/prisma#16851] [Q&A]
### Bug description

Below are some very serious bugs that need to get resolved asap.

I can't use select with groupby in Prisma which is very frustrating.

Next, if using a raw query for it and then can't get the count of records properly, it always appends n as a suffix.

Can't append dynamic where clause when making the raw query.

These things should be improved in Prisma.

### Prisma information

```
Environment variables loaded from .env
prisma                  : 4.7.1
@prisma/client          : 4.3.1
Current platform        : debian-openssl-1.1.x
Query Engine (Node-API) : libquery-engine 272861e07ab64f234d3ffc4094e32bd61775599c (at ../../../../usr/local/lib/node_modules/prisma/node_modules/@prisma/engines/libquery_engine-debian-openssl-1.1.x.so.node)
Migration Engine        : migration-engine-cli 272861e07ab64f234d3ffc4094e32bd61775599c (at ../../../../usr/local/lib/node_modules/prisma/node_modules/@prisma/engines/migration-engine-debian-openssl-1.1.x)
Introspection Engine    : introspection-core 272861e07ab64f234d3ffc4094e32bd61775599c (at ../../../../usr/local/lib/node_modules/prisma/node_modules/@prisma/engines/introspection-engine-debian-openssl-1.1.x)
Format Binary           : prisma-fmt 272861e07ab64f234d3ffc4094e32bd61775599c (at ../../../../usr/local/lib/node_modules/prisma/node_modules/@prisma/engines/prisma-fmt-debian-openssl-1.1.x)
Format Wasm             : @prisma/prisma-fmt-wasm 4.7.1-1.272861e07ab64f234d3ffc4094e32bd61775599c
Default Engines Hash    : 272861e07ab64f234d3ffc4094e32bd61775599c
Studio                  : 0.477.0
```

### Environment & setup

Os- Linux
Database- Mysql
Nodejs- v17.8.0

### Prisma Version

    "@prisma/client": "^4.2.1",
    "prisma": "^4.7.1",

**Comment by janpio** [comment]
Can you please include in your description what is actually broken? Your description is not enough to understand what is broken for you. We will need code examples, data, anything that helps us understand what is going on.

**Comment by janpio** [comment]
> it always appends n as a suffix.

I assume the data returned is a BigInt, where this is the normal representation in Javascript. 
See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt

**Comment by mukul-123** [comment]
There are three models defined
- Group
- Contact
- ContactGroup.
It is a many to many relation

prisma.contactGroup.groupBy({
      by: ["group_id"],
      _count: {
        contact_id: true,
      },
      where: {
        group: groupWhere,
        contact: contactWhere,
      },
    });

Can you let me know is there any way that i can use select the columns of contact or group model ?

I searched on it but failed to use select in it.

Next, i used rawquery
select count(contactGroup.id) as total.

I get response as {total:1n}.

Why n is coming as suffix ?

Let me know if i can use select with group by in prisma ?

**Reply by mukul-123** [reply]
Please give me some updates.

Earlier you were asking for more description

Is this not enough?

**Reply by janpio** [reply]
Please watch your tone. You are using free software and not paying for support, so you are not entitled to my responses here in a certain amount of time.

**Reply by janpio** [reply]
> Can you let me know is there any way that i can use select the columns of contact or group model ?

I do not know, what do you mean with your question? What do you want to achieve?
If you want to group something, that would be an aggregation which is documented here: https://www.prisma.io/docs/concepts/components/prisma-client/aggregation-grouping-summarizing

> Next, i used rawquery select count(contactGroup.id) as total.
> 
> I get response as {total:1n}.
> 
> Why n is coming as suffix ?

I responded to that above: https://github.com/prisma/prisma/discussions/16851#discussioncomment-4413567
Did you read that link? Did it not explain what the `n` is for?

> Let me know if i can use select with group by in prisma ?

Is that that the same question as above? If so, I do not understand what you mean.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
